### ISSUE TITLE
`terraform_remote_state` remote backend error/bug

### ISSUE THREAD

**Issue by bensojona** [terraform-providers/terraform-provider-tfe#8]
3 different issues here.

1. Using the `terraform_remote_state` data source with a "remote" backend and setting the `workspace` block results in the error "Error: data.terraform_remote_state.producer: config (workspaces): '' expected type 'string', got unconvertible type '[]map[string]interface {}'" - I believe I'm setting it appropriately (below), but no dice
1. When manipulating the `workspace` block trying to get it to work, it will cause Terraform to crash (either removing the workspace block completely or entering values that Terraform doesn't expect)
1. Setting the terraform `remote` backend block and running a plan in TFE results in the error "Setup failed: Failed to copy slug dir: lstat /Users: no such file or directory"

### Terraform Version
```
Terraform v0.11.8
+ provider.aws v1.33.0
+ provider.vault v1.1.1
```

### Terraform Configuration Files

https://gist.github.com/bensojona/962220df94b3681731d9814c4a2d3060

You can comment out the `consumer_workspace` module if trying to reproduce, it's irrelevant to the error.

```hcl
variable "tfe_org_name"           { }
variable "tfe_producer_workspace" { }
variable "tfe_consumer_workspace" { }
variable "tfe_consumer_token"     { }
variable "ttl"                    { default = "1" }

/*
# Getting the below error in TFE when running remote plans, commenting this out for now
# "Setup failed: Failed to copy slug dir: lstat /Users: no such file or directory"
terraform {
  backend "remote" {
  }
}
*/

data "terraform_remote_state" "producer" {
  backend = "remote"

  config {
    organization = "${var.tfe_org_name}"
    token        = "${var.tfe_consumer_token}"

    workspaces {
      name = "${var.tfe_producer_workspace}"
    }
  }
}

module "consumer_workspace" {
  # source = "github.com/hashicorp/terraform-guides//infrastructure-as-code/dynamic-aws-creds/consumer-workspace"
  source = "../../dynamic-aws-creds/consumer-workspace"

  backend = "${data.terraform_remote_state.producer.backend}"
  role    = "${data.terraform_remote_state.producer.role}"
  name    = "${var.tfe_consumer_workspace}"
  ttl     = "${var.ttl}"
}

```

### Debug Output

https://gist.github.com/bensojona/2d474b421e35c6ae87451963e28771e2

### Crash Output

This crash is from attempting to manipulate the `workspace` block in `terraform_remote_state` or removing it completely, which is separate from the `terraform plan` error I get with the above config - but still an issue.

https://gist.github.com/bensojona/769c79c6ecae0bf23e86817c17e3a34c

### Expected Behavior

`terraform_remote_state` data source should have pulled remote state from workspace. I should also be able to enter incorrect workspace block config with a proper error vs. crashing.

### Actual Behavior

Errors with `Error: data.terraform_remote_state.producer: config (workspaces): '' expected type 'string', got unconvertible type '[]map[string]interface {}'`.

Or a crash if manipulating the `workspace` block.

### Steps to Reproduce

1. `terraform init`
2. `terraform plan`

**Comment by bensojona** [comment]
Based on behavior I believe I remember from the ‘terraform push’ days, I think this error is happening because the plan command is pushing up not just Terraform config, but also everything in .terraform (which has local references to /User). 

Stretching my memory, but I though using the -vcs (?) or -exclude (?) switches enabled me to be explicit about what was and wasn’t pushed up to TFE to get around this. Though those don’t seem to work for the plan command.

https://www.terraform.io/docs/commands/push.html#packaged-files

**Comment by svanharmelen** [comment]
@bensojona points 1 and 2 are going to be fixed by this PR https://github.com/hashicorp/terraform/pull/18759

Point 3 is next on the list...

**Comment by bensojona** [comment]
@svanharmelen - nice!

**Comment by svanharmelen** [comment]
@bensojona these issues should all be solved with the release of TF v0.11.10. I'll go ahead and close this issue, but feel free to reopen it (or create a new issue) if you still find any related issues when using v0.11.10.

**Comment by hatched-DavidMichon** [comment]
Were you able to fix the 3rd point: Setting the terraform remote backend block and running a plan in TFE results in the error "Setup failed: Failed to copy slug dir: lstat /Users: no such file or directory".

Experienced this with v0.11.10

**Comment by svanharmelen** [comment]
@hatched-DavidMichon we received reports about that as well, so we are looking into that. But its not related to this provider, it's purely a `remote` backend issue. This will be fixed in the upcoming v0.12.0 (alpha/beta) release.
